src/network/networkd-bridge-mdb.c

```diff
@@ -252,30 +263,46 @@ static int bridge_mdb_verify(BridgeMDB *mdb) {
         if (section_is_invalid(mdb->section))
                 return -EINVAL;
 
-        if (mdb->family == AF_UNSPEC)
-                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                         "%s: [BridgeMDB] section without MulticastGroupAddress= field configured. "
-                                         "Ignoring [BridgeMDB] section from line %u.",
-                                         mdb->section->filename, mdb->section->line);
-
-        if (!in_addr_is_multicast(mdb->family, &mdb->group_addr))
-                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                         "%s: MulticastGroupAddress= is not a multicast address. "
-                                         "Ignoring [BridgeMDB] section from line %u.",
-                                         mdb->section->filename, mdb->section->line);
-
-        if (mdb->family == AF_INET) {
-                if (in4_addr_is_local_multicast(&mdb->group_addr.in))
+        switch (mdb->type) {
+        case BRIDGE_MDB_ENTRY_TYPE_L2:
+                if (!ether_addr_is_multicast(&mdb->l2_addr))
+                        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                "%s: MulticastGroupAddress= is not an L2 multicast address. "
+                                "Ignoring [BridgeMDB] section from line %u.",
+                                mdb->section->filename, mdb->section->line);
+                break;
+        case BRIDGE_MDB_ENTRY_TYPE_L3:
+                if (mdb->family == AF_UNSPEC)
                         return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                 "%s: MulticastGroupAddress= is a local multicast address. "
+                                                 "%s: [BridgeMDB] section without MulticastGroupAddress= field configured. "
                                                  "Ignoring [BridgeMDB] section from line %u.",
                                                  mdb->section->filename, mdb->section->line);
-        } else {
-                if (in6_addr_is_link_local_all_nodes(&mdb->group_addr.in6))
+
+                if (!in_addr_is_multicast(mdb->family, &mdb->group_addr))
                         return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                 "%s: MulticastGroupAddress= is the multicast all nodes address. "
+                                                 "%s: MulticastGroupAddress= is not a multicast address. "
                                                  "Ignoring [BridgeMDB] section from line %u.",
                                                  mdb->section->filename, mdb->section->line);
+
+                if (mdb->family == AF_INET) {
```

Comment:
mght actually be a candidate for switch statement too

Reply:
All of the L3 checks were already present and I didn't change them. Just changed their indentation to put them in the switch statement.
Changing this into a switch statement sounds like a good idea though, it gets rid of the nested if statements, which force us to use brackets to prevent "else" ambiguity. Should I do it as part of this PR?

Reply:
It might make sense to check for AF_INET6 on the "else" instead of just catching everything, and putting an "assert_not_reached" on the else case. I'm hesitant to change code behaviour that is not directly related to the L2 BridgeMDB entries, which is the topic of this PR though.